hopsworks-common/src/main/java/io/hops/hopsworks/common/hdfs/DistributedFileSystemOps.java

```diff
@@ -257,6 +260,82 @@ public FileStatus getFileStatus(Path location) throws IOException {
     return dfs.getFileStatus(location);
   }
 
+  /**
+   * Attach XAttr to file or directory in the given path
+   *
+   * @param path
+   * @return
+   * @throws IOException
+   */
+  public void setXAttr(String path, String name, byte[] value, EnumSet<XAttrSetFlag> flag) throws IOException {
+    Path hdfsPath = new Path(path);
+    dfs.setXAttr(hdfsPath, name, value, flag);
+  }
+  
+  /**
+   * Insert new XAttr value attached to file or directory in the given path
+   *
+   * @param path
+   * @return
+   * @throws IOException
+   */
+  public void insertXAttr(String path, String name, byte[] value) throws IOException {
+    Path hdfsPath = new Path(path);
+    insertXAttr(hdfsPath, name, value);
+  }
+  
+  public void insertXAttr(Path path, String name, byte[] value) throws IOException {
+    EnumSet<XAttrSetFlag> flags = EnumSet.noneOf(XAttrSetFlag.class);
+    flags.add(XAttrSetFlag.CREATE);
+    dfs.setXAttr(path, name, value, flags);
+  }
+  
+  /**
+   * Insert or update XAttr value attached to file or directory in the given path
+   *
+   * @param path
+   * @return
+   * @throws IOException
+   */
+  public void upsertXAttr(String path, String name, byte[] value) throws IOException {
+    Path hdfsPath = new Path(path);
+    upsertXAttr(hdfsPath, name, value);
+  }
+  
+  public void upsertXAttr(Path path, String name, byte[] value) throws IOException {
+    EnumSet<XAttrSetFlag> flags = EnumSet.noneOf(XAttrSetFlag.class);
+    if(dfs.getXAttr(path, name) != null) {
+      flags.add(XAttrSetFlag.REPLACE);
+    } else {
+      flags.add(XAttrSetFlag.CREATE);
+    }
+    dfs.setXAttr(path, name, value, flags);
+  }
+  
+  /**
+   * Read XAttr attached to file or directory in the given path
+   *
+   * @param path
+   * @return xattr value as byte array
+   * @throws IOException
+   */
+  public byte[] getXAttr(String path, String name) throws IOException {
```

Comment:
Can you add the `Path path` equivalent?

Reply:
It is already there:
https://github.com/AlexHopsworks/hopsworks/blob/d616696a3fbb495d22fca126556e3405cc62c710/hopsworks-common/src/main/java/io/hops/hopsworks/common/hdfs/DistributedFileSystemOps.java#L842